stdlib/private/SwiftPrivateLibcExtras/Subprocess.swift

```diff
@@ -264,9 +264,26 @@ public enum ProcessTerminationStatus : CustomStringConvertible {
 
 public func posixWaitpid(_ pid: pid_t) -> ProcessTerminationStatus {
   var status: CInt = 0
+#if CYGWIN
+  final class Box<T> {
+    let value: T
+
+    init(_ value: T) {
+      self.value = value
+    }
+  }
+
+  let unmanaged = Unmanaged.passRetained(Box<Int32>(status))
+  let statusPtr = __wait_status_ptr_t(__int_ptr: UnsafeMutablePointer<Int32>(unmanaged.toOpaque()))
+
+  if waitpid(pid, statusPtr, 0) < 0 {
+    preconditionFailure("waitpid() failed")
+  }
```

Comment:
Sorry, could you explain how does this work?  Why does the system `waitpid()` function need a Swift box?